client/src/main/java/com/networknt/client/oauth/SAMLBearerRequest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2016 Network New Technologies Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.networknt.client.oauth;
+
+
+import com.networknt.client.Http2Client;
+import com.networknt.common.DecryptUtil;
+import com.networknt.common.SecretConstants;
+import com.networknt.config.Config;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This holds values used to call the SAML Bearer grant flow from the OAuth Server.
+ * In this version client presents a SAML token and a JWT token (requirment for project)
```

Comment:
Can we remove the "requirement for project" comment here? This should be a solution against a pattern that can be used across many projects.

Reply:
Yes, i put it in there because there could be confusion about why we have a JWT token in SAMLBearer grant type. This pattern is unlikely to be widely adopted outside CIBC in my opinion. And another weakness in this approach is that there is nothing tying the SAML to the JWT, so a potential attacker could mix SAMLs with JWTs. A improved approach may be for the SAML to be part of the JWT and the entire thing signed.

Reply:
After speaking with the solution designer on this, this pattern could be used across many projects, and has application outside our team. It is a combination of publicly available RFC-7522 & RFC-7523.
For now we can assume that the security considerations have been thought through when designed, and if you find an actual case where a malicious subject or jwt can be swapped in or out, please raise it internally.
Until then, we can move forward with this design under the notion that is not a specific requirement for a project, and the comment can be updated to reflect this.